Core/include/Acts/EventData/detail/CorrectedTransformationFreeToBound.hpp

```diff
@@ -91,7 +91,7 @@ struct CorrectedFreeToBoundTransformer {
       const FreeVector& freeParams, const FreeSymMatrix& freeCovariance,
       const Surface& surface, const GeometryContext& geoContext,
       NavigationDirection navDir = NavigationDirection::Forward,
-      LoggerWrapper logger = getDummyLogger()) const;
+      const Logger& logger = getDummyLogger()) const;
```

Comment:
I guess in the best case we never end up with a dummy logger but it gets always passed down from the user, right?

Reply:
That's what I'm thinking yeah. The dummy logger is a tradeoff where we default to not logging at all. That's preferable over constructing a local logger in the function body (which wasn't actually the case in this specific instance before this PR, anyway)